app/lib/actions/katello/flatpak/scan_remote.rb

```diff
@@ -0,0 +1,64 @@
+require 'cgi'
+require 'uri'
+module Actions
+  module Katello
+    module Flatpak
+      class ScanRemote < Actions::EntryAction
+        def plan(remote, _args = {})
+          url = format_url(remote.url)
+          plan_self(:remote_id => remote.id, :url => url)
+        end
+
+        def run
+          remote = ::Katello::FlatpakRemote.find(input[:remote_id])
+          url = input[:url]
+          request_params = {
+            method: :get,
+            headers: { accept: :json },
+            url: url
+          }
+          results = RestClient::Request.execute(request_params)
+          results = JSON.parse(results)
+          repositories = results['Results']
+          repositories.each do |repository|
+            remote_repository = remote.repositories.find_or_initialize_by(name: repository['Name'])
+            remote_repository.label = remote.name + '-' + repository['Name']
+            remote_repository.flatpak_remote = remote
+            remote_repository.save!
+            repository['Images'].each do |image|
```

Comment:
Can you remind me why we need to index all of the manifest info as well? I can understand that the index as a source of truth holds valuable info, but I can't remember what we need besides the runtime info. Will we be displaying it to the user in the UI?

Going through the Pulp API is certainly less efficient if we need to use this data frequently, but I think all of this data will be available on the manifests API since it's pretty much all coming from labels. In fact -- since we're indexing label info now, it will even be in our database twice...

If we don't need all the info -- could we get away with just storing the runtime and the digest?

Reply:
We don't need to store everything. And it should be fairly simple to add something in case we need it later..Let me simplify the fields a bit so we store less.. 👍🏼